pages/deployments/deploying_to_kubernetes.md.erb

```diff
@@ -0,0 +1,222 @@
+envsubst < k8s/app.yml | kubectl apply -f -
+
+
+echo '--- \:zzz\: Waiting for rollout'
+# TODO: add the wait step
+```
+
+You can test pipeline now that everything is in place. All you need is your
+Docker image.
+
+## Test the Pipeline
+
+Open the deployment pipline and click "New Build". Click "Options" and set the
+`DOCKER_IMAGE` environment variable.
+
+<%= image "manual_deployment.png", width: 585/2, height: 661/2, alt: "New Manual Build" %>
+
+Assuming your agents have the required access, then success! :tada:
+
+<%= image "manual_deployment_success.png", width: 743/2, height: 728/2, alt: "Manual Build Success" %>
+
+## Automate Deployments
+
+We can do better than manual deployments. We'll use a [trigger
+steps](https://buildkite.com/docs/pipelines/trigger-step) to connect the test
+and deploy pipelines.
+
+Add these steps to your existing `.buildkite/pipeline.yml`. Be sure to add a
+`wait` step before the tirgger otherwise deploys will trigger at the wrong
+time and even for failed builds!
+
+The example `trigger` step creates a build with the same message, commit, and
+branch. `buildkite-agent pipeline-upload` interpolates environment variables so
+the correct values are replaced when the pipeline starts. The `env` setting
+passes along the `DOCKER_IMAGE` environment variable. Remember, this Docker
+image must be build and pushed in a subsequent step.
+
+Lastly, the `branches` options indicates to only build on `master`. This
+prevents deploying unexpected topic branches.
+
+```yml
+  # Add a wait step to only deploy after all steps complete
+  - wait
+
+  - label: ':rocket: Deploy'
+    # TODO: replace with your deploy pipeline's name
+    trigger: kubernetes-tutorial-deploy
+    # Only trigger on master build
+    build:
+      message: "${BUILDKITE_MESSAGE}"
+      commit: "${BUILDKITE_COMMIT}"
+      branch: "${BUILDKITE_BRANCH}"
+      env:
+        # TODO: replace with your Docker image name
+        DOCKER_IMAGE: "asia.gcr.io/buildkite-kubernetes-tutorial/app:${BUILDKITE_BUILD_NUMBER}"
+    branches: master
+```
+
+It's magic time. Push some code. :tada: Continuous deployment!
+
+<%= image "final_test_pipeline.png", width: 736, height: 199, alt: 'Final Test Pipeline' %>
+
+## Next Steps
+
+Congratulations! :tada: You've setup a continuous deployment pipeline to
+Kubernetes. Practically speaking there are some things to do next. You could
+add a [block step](https://buildkite.com/docs/pipelines/block-step) before the
+trigger to enforce manual deploys on topic branches for example. You can also
+use [Github's Deployment API](https://buildkite.com/blog/github-deployments) to
+trigger deployments as well. You'll also want to expose your application to the
+internet with a [Kubernetes
+Service](https://kubernetes.io/docs/concepts/services-networking/service/).
+Lastly, generating manifests with `envsubst` only gets you so far. Projects
+like [kustomize](https://kustomize.io/) are better suited for this use case.
```

Comment:
It would be good to have some more structure to this paragraph. Maybe have a : after 'things to do next' then have dot points with the content you've already got?